app/assets/javascripts/discourse/app/controllers/preferences/profile.js

```diff
@@ -91,8 +91,7 @@ export default class ProfileController extends Controller {
     return siteFields
       .filter(
         (siteField) =>
-          siteField.requirement === "for_all_users" &&
-          isEmpty(userFields[siteField.id])
+          siteField.requirement === "for_all_users" && !userFields[siteField.id]
```

Comment:
Fields of type `confirmation` are automatically cast from `null` to `false`. This addresses that.